### PR TITLE
Add cycstub command that generates archetype stub directories

### PR DESCRIPTION
**Pull request by rwcarlsen**
This will replace cycstub.  The cycstub command allows users to give a archetype type and libname and name and it generates a folder full of everything needed to python install.py the archetype - including finding cyclus and all necessary deps.  It even allows the user to append in new archetypes to an existing stub lib directory.  Travis-CI doesn't work with the stub test we added due to custom cyclus install location and cyclus deps finding issues in the custom environment.  Whence the shift to circleci and docker.

### CODE REVIEW

**Comment by rwcarlsen** [comment]
This will need a rebase once #1203 goes in for CI to pass.

**Comment by gidden** [comment]
time for a rebase

File: cli/cycstub [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+import shutil
+import os.path
+import re
+
+def rename_cmake(files, libpath, libname, archname) :
+    archfname = fromcapcase(archname)
+    for f in files:
+        with open(f, 'r') as inp:
+            s = inp.read()
+        s = s.replace('@archetype_spec@', '{0}:{1}:{2}'.format(libpath, libname, archname))
```
Comment (gidden) [inline cli/cycstub]: line length

File: cli/cycstub [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+import argparse
+import subprocess
+import sys
+import os
+import shutil
+import os.path
+import re
+
+def rename_cmake(files, libpath, libname, archname) :
```
Comment (gidden) [inline cli/cycstub]: spacing before colon

**Comment by gidden** [comment]
I'll fix up some pep8 issues. This could probably use another pair of eyes since it was worked on heavily by @rwcarlsen and myself. Would @gonuke, @scopatz, @katyhuff, @mbmcgarry, @opotowsky or others care to take a look?

**Comment by gidden** [comment]
ok, should be ready to go

**Comment by rwcarlsen** [comment]
Merging in ~1 hr if nobody else chimes in.

**Pull request merged**